scipy/_lib/tests/test_public_api.py

```diff
@@ -311,3 +314,34 @@ def check_importable(module_name):
         raise AssertionError("Modules that are not really public but looked "
                              "public and can not be imported: "
                              "{}".format(module_names))
+
+
+@pytest.mark.parametrize("module_name",
+                         ['scipy.stats.stats',
+                          'scipy.stats.morestats'])
+def test_private_but_present_deprecation(module_name):
+    # gh-18279, gh-17572, gh-17771 noted that deprecation warnings
+    # for imports from private modules
+    # were misleading. Check that this is resolved.
+    module = import_module(module_name)
+    sub_module_name = module_name.split(".")[1]
+    sub_module = import_module(f"scipy.{sub_module_name}")
```

Comment:
Can we use consistent naming here with the updated function signature? I.e. `sub_package[_name]` instead of `sub_module`